src/Swashbuckle.AspNetCore.SwaggerUI/SwaggerUIMiddleware.cs

```diff
@@ -65,12 +74,16 @@ public async Task Invoke(HttpContext httpContext)
         {
             var httpMethod = httpContext.Request.Method;
 
+            var routePrefix = _options.GetDynamicRoutePrefix is not null ?
+                                    _options.GetDynamicRoutePrefix(httpContext) :
+                                    _options.RoutePrefix;
+
             if (HttpMethods.IsGet(httpMethod))
             {
                 var path = httpContext.Request.Path.Value;
 
                 // If the RoutePrefix is requested (with or without trailing slash), redirect to index URL
-                if (Regex.IsMatch(path, $"^/?{Regex.Escape(_options.RoutePrefix)}/?$", RegexOptions.IgnoreCase))
+                if (Regex.IsMatch(path, $"^/?{Regex.Escape(routePrefix)}/?$", RegexOptions.IgnoreCase))
```

Comment:
I was thinking more something like `Func<HttpContext, bool>` or `Func<HttpRequest, bool>` to match against, then it doesn't necessarily have to be something to do with Regex if the user wants something custom.